src/service/cid-impl.js

```diff
@@ -118,13 +124,64 @@ export class Cid {
         'The CID scope and cookie name must only use the characters ' +
         '[a-zA-Z0-9-_.]+\nInstead found: %s',
         getCidStruct.scope);
-    return consent.then(() => {
-      return viewerForDoc(this.ampdoc).whenFirstVisible();
-    }).then(() => {
-      return getExternalCid(this, getCidStruct,
-          opt_persistenceConsent || consent);
+    return isOptedOutOfCid(this.ampdoc).then(optedOut => {
+      // TODO(aghassemi): null or '' ?
+      if (optedOut) {
+        return null;
+      } else {
+        return consent.then(() => {
+          return viewerForDoc(this.ampdoc).whenFirstVisible();
+        }).then(() => {
+          return getExternalCid(this, getCidStruct,
+              opt_persistenceConsent || consent);
+        });
+      }
     });
   }
+
+  /**
+   * User will be opted out of Cid issuance for all scopes.
+   * When opted-out Cid service will reject all `get` requests.
+   *
+   * @return {!Promise}
+   */
+  optOut() {
+    return optOutOfCid(this.ampdoc);
+  }
+}
+
+/**
+ * User will be opted out of Cid issuance for all scopes.
+ * When opted-out Cid service will reject all `get` requests.
+ *
+ * @return {!Promise}
+ * @visibleForTesting
+ */
+export function optOutOfCid(ampdoc) {
+
+  // Tell the viewer that user has opted out.
+  viewerForDoc(ampdoc).sendMessage(CID_OPTOUT_VIEWER_MESSAGE);
```

Comment:
does it make sense to move this little thing into cid-impl? so we have centralized place managing the CID APIs